transmogrifier/sources/transformer.py

```diff
@@ -229,67 +222,36 @@ def parse_source_file(cls, source_file: str) -> Iterator[dict[str, JSON] | Tag]:
         """
 
     @final
-    def _transform(
-        self, source_record: dict[str, JSON] | Tag
-    ) -> timdex.TimdexRecord | None:
+    def transform(self, source_record: dict[str, JSON] | Tag) -> timdex.TimdexRecord:
         """
-        Private method called for both XML and JSON transformations, where
-        all logic is shared except source_record type.
+        Transform source record into TimdexRecord instance.
+
+        Instantiates a TimdexRecord instance with required fields and runs fields methods
+        for optional fields. The optional field methods return values or exceptions that
+        prompt the __next__ method to skip the entire record.
```

Comment:
Very nice, thanks for adding!

One request: maybe a final sentence explaining that after optional fields are set, some shared-across-sources derived fields are also attempted from required + optional field values set by the source transformer?

Reply:
Good call!

Reply:
@ehanson8 Has this change been implemented? 🤔

Reply:
Not yet, I was saving the commit to include your suggestions as well!